components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```
@@ -67,202 +71,188 @@
-                consumerKey = httpRequest.getParameter(OAuth.OAUTH_CLIENT_ID);
+        /*Since oltu library sends OAthProblemException upon real exception and input errors we need to show
+            input errors when debugging and need to show the error logs when real exception thrown
+        */
+        if (OAuthError.TokenResponse.INVALID_REQUEST.equalsIgnoreCase(e.getError()) ||
+                OAuthError.TokenResponse.UNSUPPORTED_GRANT_TYPE.equalsIgnoreCase(e.getError())) {
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid request or unsupported grant type: " + e.getError() + ", description: " +
+                        e.getDescription());
             }
+        } else {
+            log.error("Error while creating the Carbon OAuth token request", e);
+        }
 
-            if (StringUtils.isNotEmpty(consumerKey)) {
-                OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
-                try {
-                    String appState = oAuthAppDAO.getConsumerAppState(consumerKey);
-                    if (StringUtils.isEmpty(appState)) {
-                        if (log.isDebugEnabled()) {
-                            log.debug("A valid OAuth client could not be found for client_id: " + consumerKey);
-                        }
-                        OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse
-                                .SC_UNAUTHORIZED)
-                                .setError(OAuth2ErrorCodes.INVALID_CLIENT)
-                                .setErrorDescription("A valid OAuth client could not be found for client_id: " +
-                                        consumerKey).buildJSONMessage();
-                        return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody())
-                                .build();
-                    }
+        throw new TokenEndpointBadRequestException(null, null, e);
```

Can we create a new constructor which takes Throwable instead of passing null values?